Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
Here are alternative names for the `ThreadHandover` type:

* Thread{Box,Wrapper,Package,Container}
* CrossThread{Box,Wrapper,Package,Container}
* ThreadTransfer{Box,Wrapper,Package,Container}
* ThreadSafe{Box,Wrapper,Package,Container}

And instead of export/import methods, you'd just have an `init(items:)` initializer and an `importItems()`/`unboxItems()` throwing function.

For example:

```swift
class ThreadBox<T: ThreadConfined> {
  init(items: [T])
  func unboxItems() -> throws [T]
}
```